src/decoder/stream.rs

```diff
@@ -777,16 +749,16 @@ impl StreamingDecoder {
             ImageData(type_str) => {
                 debug_assert!(type_str == IDAT || type_str == chunk::fdAT);
                 let len = std::cmp::min(buf.len(), self.current_chunk.remaining as usize);
-                let buf = &buf[..len];
-                let consumed = self.inflater.decompress(buf, image_data)?;
-                self.current_chunk.crc.update(&buf[..consumed]);
-                self.current_chunk.remaining -= consumed as u32;
+                if !self.decode_options.ignore_crc {
```

Comment:
This check should be added regardless of this PR. We waste time computing image data CRCs even if CRC checks are disabled...